tokio-executor/src/global.rs

```diff
@@ -131,6 +171,66 @@ pub fn spawn<T>(future: T)
         .unwrap()
 }
 
+/// Submits a future for execution on the default executor, returning a
+/// [`SpawnHandle`] that allows access to the result of the spawned future
+/// and can cancel it if it is no longer necessary.
+///
+/// Spawning the future behaves similarly to the [`spawn`] function, but with
+/// the addition of returning a `SpawnHandle`. A `SpawnHandle` is itself a
+/// future which will eventually complete with the value returned by the
+/// spawned future.
+///
+/// If the spawned future is no longer needed, the [`SpawnHandle::cancel`]
+/// method will cancel it. Dropping the `SpawnHandle` will *not* cancel the
+/// spawned future, but will result in the item returned by the spawned future
+/// being discarded.
+///
+/// # Panics
+///
+/// This function will panic if the default executor is not set or if spawning
+/// onto the default executor returns an error.
+///
+/// # Examples
+///
+/// ```rust
+/// # extern crate futures;
+/// # extern crate tokio_executor;
+/// # use tokio_executor::{spawn_handle, SpawnHandle};
+/// use futures::Future;
+/// use futures::future::lazy;
+///
+/// # pub fn dox() {
+/// let handle: SpawnHandle<&'static str, ()> = spawn_handle(lazy(|| {
+///     Ok("hello from the future!")
+/// }));
+/// assert_eq!(handle.wait().unwrap(), "hello from the future!");
+/// # }
+/// # pub fn main() {}
+/// ```
+///
+/// [`SpawnHandle`]: struct.SpawnHandle.html
+/// [`spawn`]: fn.spawn.html
+/// [`SpawnHandle::cancel`]: struct.SpawnHandle.html#method.cancel
+pub fn spawn_handle<T>(future: T) -> SpawnHandle<T::Item, T::Error>
```

Comment:
The `T` here was for consistency with the definition of `spawn` --- I think the struct used `F` because earlier in the implementation it also had `T` and `E` type parameters which ended up being unnecessary. Fixed it in b55e9e9b078ce6af6cf9511a0ecd3596b79a5988.